tools/sslsniff.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python
+#
+# sslsniff  Captures data on SSL_READ or SSL_WRITE functions of OpenSSL
+#           For Linux, uses BCC, eBPF.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# 12-Aug-2016    Adrian Lopez   Created this.
+# 13-Aug-2016    Mark Drayton   Fix SSL_Read
+
+import ctypes as ct
+from bcc import BPF
+
+prog = """
+#include <linux/ptrace.h>
+#include <linux/sched.h>        /* For TASK_COMM_LEN */
+
+struct probe_SSL_data_t
+{
+        u64 timestamp_ns;
+        u32 pid;
+        char comm[TASK_COMM_LEN];
+        char v0[476];
+};
+
+BPF_PERF_OUTPUT(perf_SSL_write);
+
+int probe_SSL_write(struct pt_regs *ctx)
```

Comment:
In my quick test it appears that listing args on the function definition works, e.g.:

    int probe_SSL_write(struct pt_regs *ctx, void *ssl, void *buf, int num) {

(same for the other probes)

This is (IMO) clearer than using the PT_REGS_* macros to access the function args.